app/javascript/src/tooltips.js

```diff
@@ -1,29 +1,37 @@
 // Helper tooltips.
-const activateTooltips = () => {
+export const activateTooltips = () => {
   // Regular spans
-  $('.daria-tooltip').tooltip();
-
+  $(".daria-tooltip").tooltip();
   // Some evil magic for things attached to labels with rails-bootstrap-form
-  $('.tooltip-header-input').append(' <span class="tooltip-header-help">(?)</span>');
-  $('.tooltip-header-input').parent('.form-group' ).find('.form-control').each((_, x) => {
-    $(x).parents( '.form-group' ).find( '.tooltip-header-help' ).tooltip({
-      html: true,
-      placement: 'bottom',
-      title: $(x).data('tooltip-text')
+  $(".tooltip-header-input:not(:has(span.tooltip-header-help))").append(
+    ' <span class="tooltip-header-help">(?)</span>'
+  );
+  $(".tooltip-header-input")
+    .parent(".form-group")
+    .find(".form-control")
+    .each((_, x) => {
+      $(x)
+        .parents(".form-group")
+        .find(".tooltip-header-help")
+        .tooltip({
+          html: true,
+          placement: "bottom",
+          title: $(x).data("tooltip-text"),
+        });
     });
-  });
 
   // Similarly evil magic for checkboxes
-  $('.tooltip-header-checkbox').append(' <span class="tooltip-header-help">(?)</span>')
-  $('.tooltip-header-checkbox').each((_, x) => {
-    let text = $(x).parent().find('input[type=checkbox]').data('tooltip-text');
-    $(x).parent().find( 'span.tooltip-header-help' ).tooltip({
+  $(".tooltip-header-checkbox:not(:has(span.tooltip-header-help))").append(
+    ' <span class="tooltip-header-help">(?)</span>'
+  );
+  $(".tooltip-header-checkbox").each((_, x) => {
+    let text = $(x).parent().find("input[type=checkbox]").data("tooltip-text");
+    $(x).parent().find("span.tooltip-header-help").tooltip({
       html: true,
-      placement: 'bottom',
+      placement: "bottom",
       title: text,
     });
```

Comment:
are the two "evil" comments still valid here, or have we adjusted it to be evil magic that works with bootstrap form and avoids the React?

Reply:
That's a really good question. The "evil" comments are still valid b/c I haven't removed all of our bootstrap form tooltips. I've just added the ability to use React to render tooltips and have them behave in the same way. 

If we eventually get all the tooltips moved to React and don't use bootstrap form, we could probably refactor this code away.

Aside: It looks like my JS formatting extension was a little assertive here and autoformatted this. I'll undo that so the diff is super clear.